src/storagetm/storagetm.c

```diff
@@ -950,9 +950,14 @@ static int device_added(Context *c, sd_device *device) {
                 .st_mode = S_IFBLK,
         };
 
-        r = sd_device_get_devnum(device, &lookup_key.st_rdev);
+        /* MIPS OABI declares st_rdev as unsigned long instead of dev_t.
+         * Use a temp var to avoid passing an incompatible pointer.
+         * https://sourceware.org/bugzilla/show_bug.cgi?id=21278 */
+        dev_t devnum;
+        r = sd_device_get_devnum(device, &devnum);
```

Comment:
Not strong opinion, but how about the following?
```c
assert_cc(sizeof(dev_t) == sizeof(lookup_key.st_rdev));
r = sd_device_get_devnum(device, (dev_t*) &lookup_key.st_rdev);
```

Reply:
dev_t and st_rdev have different sizes on MIPS32 OABI, so that assert would be counter-productive.

Reply:
If necessary, we could add a runtime assert to ensure that the devnum value will fit in st_rdev. But I'm not really sure if that's needed?

Reply:
Oh, then, does the fix really work??
We push `NvmeSubsystem` object to `Context.subsystems` hashmap with `struct stat` of the device node obtained by `fstat()` as the key.
But when finding the object, we construct a key by `sd_device_get_devnum()`...

Reply:
I really have no clue how to test if it "really works". This is not a systemd component I am familiar with and I don't have a MIPS system to test on myself.

I think it should work fine so long as sd_device_get_devnum never gives us a device number that exceeds ULONG_MAX (4294967295).

Reply:
From bits/sysmacros.h in glibc:
```
/* dev_t in glibc is a 64-bit quantity, with 32-bit major and minor numbers.
   Our default encoding is MMMM Mmmm mmmM MMmm, where M is a hex digit of
   the major number and m is a hex digit of the minor number.  This is
   downward compatible with legacy systems where dev_t is 16 bits wide,
   encoded as MMmm.  It is also downward compatible with the Linux kernel,
   which (as of 2016) uses 32-bit dev_t, encoded as mmmM MMmm.

   Systems that use an incompatible encoding for dev_t should override this
   file in the appropriate sysdeps subdirectory.  */
```
Assuming the kernel still only uses 32-bit device numbers, it should be safe to ignore the upper 32-bits of a 64-bit dev_t value.

Reply:
> Assuming the kernel still only uses 32-bit device numbers, it should be safe to ignore the upper 32-bits of a 64-bit dev_t value.

OK. I agree with that. Thanks.